mono/mini/driver.c

```diff
@@ -2556,38 +2557,66 @@ mono_jit_set_aot_only (gboolean val)
 	mono_aot_only = val;
 }
 
-/**
- * mono_jit_set_aot_mode:
- */
-void
-mono_jit_set_aot_mode (MonoAotMode mode)
+static void
+mono_runtime_set_execution_mode (MonoEEMode mode)
 {
-	/* we don't want to set mono_aot_mode twice */
-	g_assert (mono_aot_mode == MONO_AOT_MODE_NONE);
-	mono_aot_mode = mode;
+	memset (&mono_ee_features, 0, sizeof (mono_ee_features));
```

Comment:
we should clear all the boolean flags here too, but I guess in the long-term we will consolidate them into the struct so it isn't a blocker for merging

Reply:
Hopefully we'll just remove all boolean flags and this won't be an issue.